src/search_request.c

```diff
@@ -143,15 +169,22 @@ RSSearchRequest *ParseRequest(RedisSearchCtx *ctx, RedisModuleString **argv, int
   }
 
   // parse the id filter arguments
-  long long numFilteredIds = 0;
-  IdFilter idf = (IdFilter){.ids = NULL, .keys = NULL};
-  RMUtil_ParseArgsAfter("INKEYS", &argv[2], argc - 2, "l", &numFilteredIds);
-  if (numFilteredIds > 0 && numFilteredIds < argc - 3) {
-
-    RedisModule_Log(ctx->redisCtx, "debug", "Filtering %d keys", numFilteredIds);
-    int pos = RMUtil_ArgIndex("INKEYS", argv, argc);
-    req->idFilter =
-        NewIdFilter(&argv[pos + 2], MIN(argc - pos - 2, numFilteredIds), &ctx->spec->docs);
+  if ((vargs = getLengthArgs("INKEYS", &nargs, argv, argc, 2))) {
+    req->idFilter = NewIdFilter(vargs, nargs, &ctx->spec->docs);
+  }
+
+  // parse RETURN argument
+  if ((vargs = getLengthArgs("RETURN", &nargs, argv, argc, 2))) {
+    if (!nargs) {
+      req->flags |= Search_NoContent;
+    } else {
+      req->retfields = malloc(sizeof(*req->retfields) * nargs);
+      req->nretfields = nargs;
+      for (size_t ii = 0; ii < nargs; ++ii) {
+        RedisModule_RetainString(ctx->redisCtx, vargs[ii]);
```

Comment:
I don't see any explicit copying of an existing RedisModuleString though. So I don't think it makes the code non-uniform here.